File: fem_meshing/triangle_read.f90
subroutine triangle_read ( infile, m, n, table )

  integer ( kind = 4 ) m, n
  integer ( kind = 4 ) i
  character ( len = * ) infile
  integer ( kind = 4 ) inunit
  integer ( kind = 4 ) table(m,n)

  open ( unit = inunit, file = infile, status = 'old' )
  do i = 1, m
    read ( inunit, * ) table(i,1:n)
  end do
  close ( unit = inunit )

  return
end

File: fem_meshing/triangle_read.cpp
int **triangle_read ( string infile, int m, int n )
{
  ifstream inunit;
  int i, j;
  int **table;

  inunit.open ( infile.c_str ( ) );

  table = new int[m][n];

  for ( i = 0; i < m; i++ )
  {
    for ( j = 0; j < n; j++ )
    {
      inunit >> table[i][j];
    }
  }
  
  inunit.close ( );
  
  return table;
}
